File: cml-compiler/cml-generator/src/test/resources/templates/lang/java/expressions/organization.java
class Organization
{
    private static Employment employment;

    private final String name;

    public Organization(String name, List<Employee> employees)
    {
        this.name = name;

        employment.linkMany(this, employees);
    }

    public String getName()
    {
        return this.name;
    }

    public List<Employee> getEmployees()
    {
        return employment.employeesOf(this);
    }

    public List<String> getEmployeeNames()
    {
        return getEmployees()
                   .stream()
                   .map(employee -> employee.getName())
                   .collect(toList());
    }

    public List<String> getEmployerNames()
    {
        return getEmployees()
                   .stream()
                   .map(employee -> employee.getEmployer())
                   .map(organization -> organization.getName())
                   .collect(toList());
    }

    public List<Organization> getEmployers()
    {
        return getEmployees()
                   .stream()
                   .map(employee -> employee.getEmployer())
                   .collect(toList());
    }

    public List<Integer> getEmployeeNumbers()
    {
        return getEmployees()
                   .stream()
                   .map(employee -> employee.getNumber())
                   .collect(toList());
    }

    public String toString()
    {
        return new StringBuilder(Organization.class.getSimpleName())
                   .append('(')
                   .append("name=").append(String.format("\"%s\"", getName())).append(", ")
                   .append("employeeNames=").append(getEmployeeNames()).append(", ")
                   .append("employerNames=").append(getEmployerNames()).append(", ")
                   .append("employeeNumbers=").append(getEmployeeNumbers())
                   .append(')')
                   .toString();
    }

    static void setEmployment(Employment association)
    {
        employment = association;
    }

    static
    {
        Employment.init(Organization.class);
    }
}

File: cml-compiler/cml-generator/src/test/resources/templates/lang/py/expressions/organization.py
class Organization:

    _employment = _Employment()

    def __init__(self, name: 'str', employees: 'List[Employee]') -> 'None':
        self.__name = name

        self._employment.link_many(self, employees)

    @property
    def name(self) -> 'str':
        return self.__name

    @property
    def employees(self) -> 'List[Employee]':
        return self._employment.employees_of(self)

    @property
    def employee_names(self) -> 'List[str]':
        return list(
            map(
                lambda employee: employee.name,
                self.employees
            )
        )

    @property
    def employer_names(self) -> 'List[str]':
        return list(
            map(
                lambda organization: organization.name,
                map(
                    lambda employee: employee.employer,
                    self.employees
                )
            )
        )

    @property
    def employers(self) -> 'List[Organization]':
        return list(
            map(
                lambda employee: employee.employer,
                self.employees
            )
        )

    @property
    def employee_numbers(self) -> 'List[int]':
        return list(
            map(
                lambda employee: employee.number,
                self.employees
            )
        )

    def __str__(self) -> 'str':
        return "%s(name=%s, employee_names=%s, employer_names=%s, employee_numbers=%s)" % (
            type(self).__name__,
            self.name,
            self.employee_names,
            self.employer_names,
            self.employee_numbers
        )